src/Microsoft.TemplateSearch.TemplateDiscovery/PackChecking/PackSourceChecker.cs

```diff
@@ -30,65 +30,89 @@ public PackSourceChecker(IPackProvider packProvider, PackPreFilterer packPreFilt
         public async Task<PackSourceCheckResult> CheckPackagesAsync()
         {
             List<PackCheckResult> checkResultList = new List<PackCheckResult>();
+            HashSet<IPackInfo> alreadySeenPacks = new HashSet<IPackInfo>();
```

Comment:
We learn the number of packs early on. It may be good to use that number to pre-allocate some capacity for this when instantiating (also applies to `checkResultList`, right?).